src/transaction/transaction.cpp

```diff
@@ -24,6 +24,19 @@ Transaction::Transaction(main::ClientContext& clientContext, TransactionType tra
     currentTS = common::Timestamp::getCurrentTimestamp().value;
 }
 
+Transaction::Transaction(TransactionType transactionType) noexcept
+    : type{transactionType}, ID{DUMMY_TRANSACTION_ID}, startTS{DUMMY_START_TIMESTAMP},
+      commitTS{common::INVALID_TRANSACTION}, clientContext{nullptr}, undoBuffer{nullptr},
+      forceCheckpoint{false} {
+    currentTS = common::Timestamp::getCurrentTimestamp().value;
+}
+Transaction::Transaction(TransactionType transactionType, common::transaction_t ID,
+    common::transaction_t startTS) noexcept
+    : type{transactionType}, ID{ID}, startTS{startTS}, commitTS{common::INVALID_TRANSACTION},
+      clientContext{nullptr}, undoBuffer{nullptr}, forceCheckpoint{false} {
+    currentTS = common::Timestamp::getCurrentTimestamp().value;
```

Comment:
Not sure how these lines are missing coverage if the DUMMY_CHECKPOINT_TRANSACTION uses them. Maybe it doesn't check code running in the global scope?

Reply:
`DUMMY_CHECKPOINT_TRANSACTION` should've used this interface. maybe it's because `DUMMY_CHECKPOINT_TRANSACTION` is static?